var/spack/repos/builtin/packages/mofem-fracture-module/package.py

```diff
@@ -0,0 +1,97 @@
+##############################################################################
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+import os
+
+
+class MofemFractureModule(CMakePackage):
+    """mofem fracture module"""
+
+    homepage = "http://mofem.eng.gla.ac.uk"
+    url = "https://bitbucket.org/likask/mofem_um_fracture_mechanics"
+
+    maintainers = ['likask']
+
+    version('0.9.42',
+        git='https://bitbucket.org/likask/mofem_um_fracture_mechanics',
+        tag='v0.9.42')
+    version('develop',
+        git='https://bitbucket.org/likask/mofem_um_fracture_mechanics',
+        branch='develop')
+
+    variant('copy_user_modules', default=True,
+        description='Copy user modules directory instead linking')
+
+    extends('mofem-cephas')
+    depends_on("mofem-users-modules", type=('build', 'link', 'run'))
+
+    @property
+    def root_cmakelists_dir(self):
+        """The relative path to the directory containing CMakeLists.txt
+
+        This path is relative to the root of the extracted tarball,
+        not to the ``build_directory``. Defaults to the current directory.
+
+        :return: directory containing CMakeLists.txt
+        """
+        spec = self.spec
+        return os.path.join(spec['mofem-cephas'].prefix.users_modules)
```

Comment:
How exactly does this work? The `CMakeLists.txt` for `mofem-cephas` is copied to its installation directory, and you run it but tell it where the user module lives and where to install it? Very clever, I like it!

P.S. `os.path.join` isn't needed here.

Reply:
Your are right. 

The CMakeLists.txt installed with mofem-cephas package set cmake environment to install extension from extension repository. It searches for modules in user provides paths, for example in Spack source path. Also it finds all cmake exported targets installed in *lib* directory, which are built with dependent extensions, f.e. mofem-users-modules or others if needed.

Reply:
> The CMakeLists.txt installed with mofem-cephas package set cmake environment to install extension from extension repository. It searches for modules in user provides paths, for example in Spack source path. Also it finds all cmake exported targets installed in lib directory, which are built with dependent extensions, f.e. mofem-users-modules or others if needed.

@likask maybe add exactly this as in-code comment, i.e. 
```
# The CmakeList.txt ....
# ....
# blablabla
```
instead of the standard doc-string?